montblanc/impl/biro/v2/loaders/loaders.py

```diff
@@ -59,8 +82,11 @@ def load(self, solver, **kwargs):
         solver.set_ref_wave(montblanc.constants.C/tf.getcol('REF_FREQUENCY')[0])
 
         # Get the baseline antenna pairs and correct the axes
-        ant1 = tm.getcol('ANTENNA1').reshape(ntime,nbl)
-        ant2 = tm.getcol('ANTENNA2').reshape(ntime,nbl)
+        ant1 = tm.getcol('ANTENNA1').reshape(file_ant_shape)
+        ant2 = tm.getcol('ANTENNA2').reshape(file_ant_shape)
+	if (data_order == ORDER_BL_TIME):
+		ant1.transpose(ant_transpose)
+		ant2.transpose(ant_transpose)
```

Comment:
@marziarivi I don't think the following lines do an in place transpose. As far as I know, they return a transposed view/copy of the array.

```python
if (data_order == ORDER_BL_TIME):
    ant1.transpose(ant_transpose)
    ant2.transpose(ant_transpose)

expected_ant_shape = (ntime, nbl)
```

They're not changed in fact, because the `expected_ant_shape` check always succeeds! This needs to be fixed.